ethcore/src/client/client.rs

```diff
@@ -139,8 +139,8 @@ impl<V> Client<V> where V: Verifier {
 		state_path.push("state");
 
 		let engine = Arc::new(try!(spec.to_engine()));
-		let mut state_db = JournalDB::from_prefs(state_path.to_str().unwrap(), config.prefer_journal);
-		if state_db.is_empty() && engine.spec().ensure_db_good(&mut state_db) {
+		let mut state_db = Box::new(OptionOneDB::from_prefs(state_path.to_str().unwrap(), config.prefer_journal));
```

Comment:
Small factory for those DB will be needed like `JournalDB::new(JorunalDBType::OptionOne)`